bufferserver/src/main/java/com/datatorrent/bufferserver/server/Server.java

```diff
@@ -199,6 +199,13 @@ private void handlePurgeRequest(PurgeRequestTuple request, final AbstractLengthP
     }
   }
 
+  public void purge(long windowId)
+  {
+    for (DataList dataList: publisherBuffers.values()) {
```

Comment:
The code is not thread safe as purge() will now be called from a different thread.